### PR TITLE
Dark mode switch refactor

### PR DESCRIPTION
**Pull request by felix-ico**
tried cleaning up the switch component a bit, pls let me know if it looks ok

### CODE REVIEW

**Review by acstll — APPROVED** [review]
Just ran this locally and it looks fine 👍

Was the clean up only intended for code, or are there any visual fixes involved?

**Comment by felix-ico** [comment]
>Was the clean up only intended for code, or are there any visual fixes involved?

Only code. Before it was a bit messy in that some css rules would use --telekom vars directly, sometimes they would overwrite local scoped vars so it was a bit hard to follow what's going on. Also there were scoped variables that weren't used at all or where the name didn't really make any sense.. some outdated comments as well

**Comment by maomaoZH** [comment]
I left one comment. The rest LGTM.

**Comment by maomaoZH** [comment]
I left one comment. The rest LGTM.

**Comment by felix-ico** [comment]
just for more context, i started updating the code "as it was before" to implement some QA fixes but had a hard time following what was going on with the different ellement states so that's why i tried to clean up the css a bit.

File: packages/components/src/components/switch/switch.css [hunk 1]
```diff
@@ -142,7 +152,7 @@
   input:focus-visible:not(:checked)
   ~ .switch__wrapper {
   box-shadow: var(--box-shadow-focus);
-  border-color: var(--telekom-color-ui-outlined-border-hovered);
+  border-color: var(--border-hover);
```
[the remark this answers is not in the document]
Comment (maomaoZH) [reply packages/components/src/components/switch/switch.css:155]: sry I just noticed the whole time the review was not submitted. Now you see it. that is good!
